### ISSUE TITLE
Multi-thread push/pull

### ISSUE THREAD

**Issue by mikelei8291** [odeke-em/drive#1000]
When I tried to push a lot of small files (size~1M), it cost more time to finish than a whole file at same size.

If drive push/pull multiple files at the same time, it will reach a higher speed.

**Comment by mikelei8291** [comment]
Oh, the 1000th issue!

**Comment by Berkmann18** [comment]
Yeah, it would be about time to have a multithreaded approach (especially as languages have the capability of seeing how cores are available on a machine).
